pkg/ccl/changefeedccl/encoder.go

```diff
@@ -492,21 +493,26 @@ func (e *confluentAvroEncoder) register(
 		return 0, err
 	}
 
-	// TODO(someone): connect the context to the caller to obey
-	// cancellation.
-	resp, err := httputil.Post(ctx, url.String(), confluentSchemaContentType, &buf)
-	if err != nil {
-		return 0, err
-	}
-	defer resp.Body.Close()
-	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
-		body, _ := ioutil.ReadAll(resp.Body)
-		return 0, errors.Errorf(`registering schema to %s %s: %s`, url.String(), resp.Status, body)
-	}
-	var res confluentSchemaVersionResponse
-	if err := gojson.NewDecoder(resp.Body).Decode(&res); err != nil {
+	var id int32
```

Comment:
Another option here is to mark the error as retriable and bubble back up to the retry loop of the job.